src/elements/AddFeatureButton/AddFeatureButton.js

```diff
@@ -53,19 +53,32 @@ const AddFeatureButton = ({ feature, tableStore, updateGroupedFeatures }) => {
   ])
 
   const addFeature = () => {
-    const groupedProjectFeatures = [
-      ...tableStore.features.groupedFeatures[
-        tableStore.features.currentProject
-      ],
-      {
-        project: feature.metadata.project,
-        featureSet: feature.metadata.name,
-        tag: feature.metadata.tag,
-        feature: feature.name,
-        alias: '',
-        originalTemplate: `${feature.metadata.project}/${feature.metadata.name}:${feature.metadata.tag}.${feature.name}`
-      }
+    const groupedProjectFeatures = tableStore.features.groupedFeatures[
+      tableStore.features.currentProject
     ]
+      ? [
+          ...tableStore.features.groupedFeatures[
+            tableStore.features.currentProject
+          ],
+          {
+            project: feature.metadata.project,
+            featureSet: feature.metadata.name,
+            tag: feature.metadata.tag,
+            feature: feature.name,
+            alias: '',
+            originalTemplate: `${feature.metadata.project}/${feature.metadata.name}:${feature.metadata.tag}.${feature.name}`
+          }
+        ]
+      : [
+          {
+            project: feature.metadata.project,
+            featureSet: feature.metadata.name,
+            tag: feature.metadata.tag,
+            feature: feature.name,
+            alias: '',
+            originalTemplate: `${feature.metadata.project}/${feature.metadata.name}:${feature.metadata.tag}.${feature.name}`
+          }
+        ]
```

Comment:
The logic is good and proper, but I think we can maybe make it more readable.
There are two parts here that repeat twice: the already-existing features, and the newly-added feature.
If we store them in their own variables, that cleans our code and reduces duplication.

```suggestion
    const existingFeatures =
      tableStore.features.groupedFeatures[tableStore.features.currentProject]
    const newFeature = {
      project: feature.metadata.project,
      featureSet: feature.metadata.name,
      tag: feature.metadata.tag,
      feature: feature.name,
      alias: '',
      originalTemplate: `${feature.metadata.project}/${feature.metadata.name}:${feature.metadata.tag}.${feature.name}`
    }

    const groupedProjectFeatures = existingFeatures
      ? [...existingFeatures, newFeature]
      : [newFeature]

    updateGroupedFeatures(groupedProjectFeatures)
```

But we can get even one step further.
If we default `existingFeatures` variable to `[]` then we don't need the conditional. And `groupedProjectFeatures` becomes redundant.

```suggestion
    const existingFeatures =
      tableStore.features.groupedFeatures[tableStore.features.currentProject] ??
      []
    const newFeature = {
      project: feature.metadata.project,
      featureSet: feature.metadata.name,
      tag: feature.metadata.tag,
      feature: feature.name,
      alias: '',
      originalTemplate: `${feature.metadata.project}/${feature.metadata.name}:${feature.metadata.tag}.${feature.name}`
    }

    updateGroupedFeatures([...existingFeatures, newFeature])
```